exporters/exporter.go

```diff
@@ -91,19 +90,6 @@ func (exporter *BaseOpenStackExporter) Describe(ch chan<- *prometheus.Desc) {
 	}
 }
 
-func (exporter *BaseOpenStackExporter) AddMetricCollectTime(collectTimeSeconds float64, metricName string, ch chan<- prometheus.Metric) {
-	metricPromatheuslabels := prometheus.Labels{
-		"openstack_service": exporter.GetName(),
-		"openstack_metric":  metricName}
-	metric := prometheus.NewDesc(
-		"openstack_metric_collect_seconds",
-		"Time needed to collect metric from OpenStack API",
-		nil,
-		metricPromatheuslabels)
-	log.Debugf("Adding metric for collecting timings: %+s", metric)
-	ch <- prometheus.MustNewConstMetric(metric, prometheus.GaugeValue, collectTimeSeconds)
-}
-
```

Comment:
Removing the specific labels, will make this fail with:


INFO[0007] Adding metric: limits_memory_used to exporter: nova  source="exporter.go:164"
2020/12/03 10:09:39 http: panic serving [::1]:60370: descriptor Desc{fqName: "openstack_metric_collect_seconds", help: "Time needed to collect metric from OpenStack API", constLabels: {}, variableLabels: [openstack_service openstack_metric]} already exists with the same fully-qualified name and const label values
goroutine 36 [running]:
net/http.(*conn).serve.func1(0xc0001a4820)
        /snap/go/current/src/net/http/server.go:1801 +0x147
panic(0x95a520, 0xc000431d10)
        /snap/go/current/src/runtime/panic.go:975 +0x47a
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0xc0000b0050, 0xc000229b10, 0x1, 0x1)
        /home/niedbalski/go/pkg/mod/github.com/prometheus/client_golang@v1.2.1/prometheus/registry.go:400 +0xad
main.probeHandler(0xab0a60, 0xc0000ca000, 0xc0001ce500)
        /home/niedbalski/GolandProjects/openstack-exporter/main.go:120 +0x6e5
net/http.HandlerFunc.ServeHTTP(0xa30530, 0xab0a60, 0xc0000ca000, 0xc0001ce500)
        /snap/go/current/src/net/http/server.go:2042 +0x44
net/http.(*ServeMux).ServeHTTP(0xdd2840, 0xab0a60, 0xc0000ca000, 0xc0001ce500)
        /snap/go/current/src/net/http/server.go:2417 +0x1ad
net/http.serverHandler.ServeHTTP(0xc0001e2000, 0xab0a60, 0xc0000ca000, 0xc0001ce500)
        /snap/go/current/src/net/http/server.go:2843 +0xa3
net/http.(*conn).serve(0xc0001a4820, 0xab2ca0, 0xc0001a0540)
        /snap/go/current/src/net/http/server.go:1925 +0x8ad
created by net/http.(*Server).Serve
        /snap/go/current/src/net/http/server.go:2969 +0x36c


Please revert the change or make sure that labels are unique for this metric.

Reply:
Fixed that, I was doing unit testing on only 1 exporter for speed sake ...